mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -455,6 +461,42 @@ def _get_job_run_id(self, run):
             return None
         return run.data.tags.get(MLFLOW_DATABRICKS_JOB_RUN_ID, None)
 
+    def _truncate_input_sources(self, run):
+        num_input_sources = len(run.inputs.dataset_inputs)
+        if num_input_sources > _MAX_LINEAGE_DATA_SOURCES:
+            _logger.warning(
+                f"Truncating the number of lineage input sources from "
+                f"{str(num_input_sources)} to 10."
+            )
+        return run.inputs.dataset_inputs[0:_MAX_LINEAGE_DATA_SOURCES]
+
+    def _get_lineage_input_sources(self, run):
+        if run is None:
+            return None
+        securable_list = []
+        if run.inputs is not None:
+            for dataset in run.inputs.dataset_inputs:
+                dataset_source = mlflow.data.get_source(dataset)
+                if (
+                    isinstance(dataset_source, DeltaDatasetSource)
+                    and dataset_source._get_source_type() == _DELTA_TABLE
+                ):
+                    # check if dataset is a uc table and then append
+                    if dataset_source.delta_table_name and dataset_source.delta_table_id:
+                        table_entity = Table(
+                            name=dataset_source.delta_table_name,
+                            table_id=dataset_source.delta_table_id,
+                        )
+                        securable_list.append(Securable(table=table_entity))
+            if len(securable_list) > _MAX_LINEAGE_DATA_SOURCES:
```

Comment:
Should `_truncate_input_sources` be called here?

Reply:
I meant to delete the method altogether.  Since we did not care about a scan or the proto generation, then it was just easier to include all of it in this method and I was able to make the ugly test that checks the truncation work.  Will update to delete the unused method.

Reply:
We immediately filter out non-delta sources, and then filter away delta sources w/o table ids, so we should only be left with valid data source by the time we're hydrating protos.

Reply:
SGTM, maybe then we can just apply the warning message update here